### PR TITLE
LWDEV-10110 Refactor the implementation of limit orders cancel operations

### PR DESCRIPTION
**Pull request by barsukovskii**
Remove obsolete class GenericLimitOrdersCanceller and all connected ones, use ExecutionContext to perform cancellation operations

### CODE REVIEW

File: src/main/kotlin/com/lykke/matching/engine/order/process/common/CancelRequest.kt [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import com.lykke.matching.engine.daos.LimitOrder
+import com.lykke.matching.engine.deduplication.ProcessedMessage
+import com.lykke.matching.engine.messages.MessageType
+import com.lykke.matching.engine.messages.MessageWrapper
+import org.apache.log4j.Logger
+import java.util.Date
+
+class CancelRequest(val limitOrders: Collection<LimitOrder>,
+                    val stopLimitOrders: Collection<LimitOrder>,
+                    val assetPairId: String?,
```
Comment (rocketmannot) [inline src/main/kotlin/com/lykke/matching/engine/order/process/common/CancelRequest.kt]: is `assetPairId` - needed ?

File: src/main/kotlin/com/lykke/matching/engine/order/process/common/LimitOrdersCancelExecutorImpl.kt [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            return executionDataApplyService.persistAndSendEvents(messageWrapper, executionContext)
+        }
+    }
+
+    private fun createAssetPairsByIdMapForOrders(orders: Collection<LimitOrder>): Map<String, AssetPair> {
+        return orders.asSequence()
+                .map { it.assetPairId }
+                .toSet()
+                .mapNotNull { assetsPairsHolder.getAssetPairAllowNulls(it) }
+                .groupBy { it.assetPairId }
```
Comment (rocketmannot) [inline src/main/kotlin/com/lykke/matching/engine/order/process/common/LimitOrdersCancelExecutorImpl.kt]: As i understand asset pair assetPairId is unique - so we dont need to perform grouping here ?

File: src/main/kotlin/com/lykke/matching/engine/order/process/common/CancelRequest.kt (hunk 1 above)
Comment (barsukovskii) [reply src/main/kotlin/com/lykke/matching/engine/order/process/common/CancelRequest.kt]: No. Fixed

File: src/main/kotlin/com/lykke/matching/engine/order/process/common/LimitOrdersCancelExecutorImpl.kt (hunk 2 above)
Comment (barsukovskii) [reply src/main/kotlin/com/lykke/matching/engine/order/process/common/LimitOrdersCancelExecutorImpl.kt]: Yes, but we group here to return map assetPairId->AssetPair

**Pull request merged**